src/git.rs

```diff
@@ -29,6 +29,42 @@ pub static DEFAULT_TYPES: Lazy<HashMap<&str, &str>> = Lazy::new(|| {
     m
 });
 
+// Singleton pattern
+pub fn get_repository(repo: &Path) -> Result<&Mutex<Repository>, Error> {
+    static REPO: OnceCell<Mutex<Repository>> = OnceCell::new();
```

Comment:
Any specific reason why this has to be a `Mutex`? Wouldn't a `once_cell::Lazy` be enough to prevent multiple initializations?

The mutex should only be needed in a multi-threaded environment 🤔